### PR TITLE
🧯 Fix contact update form

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/contacts/views.py [hunk 1]
```diff
@@ -1442,7 +1442,7 @@ def save(self, obj):
 
             Contact.get_or_create_by_urns(obj.org, self.request.user, obj.name, urns)
 
-    class Update(ModalMixin, OrgObjPermsMixin, SmartUpdateView):
+    class Update(NonAtomicMixin, ModalMixin, OrgObjPermsMixin, SmartUpdateView):
```
Comment (rowanseymour) [inline temba/contacts/views.py:1445]: Without this I'm seeing deadlocks between mailroom trying to update modified_on on the contact and (I think) the query in RP to get the contact's groups

We're still updating URNs on the RP side here but pretty soon all the updating will be done by mailroom in its own transaction so no need to have a transaction here.

File: temba/contacts/views.py [hunk 2]
```diff
@@ -1511,7 +1511,17 @@ def form_valid(self, form):
...
+            messages.success(self.request, self.derive_success_message())
+
+            response = self.render_to_response(
+                self.get_context_data(
+                    form=form,
+                    success_url=self.get_success_url(),
+                    success_script=getattr(self, "success_script", None),
+                )
+            )
+            response["Temba-Success"] = self.get_success_url()
```
Comment (rowanseymour) [inline temba/contacts/views.py:1523]: the modal isn't currently closing 🤦

Comment (ericnewcomer) [reply temba/contacts/views.py:1523]: How do you repro that? I'm not seeing that.

Comment (rowanseymour) [reply temba/contacts/views.py:1523]: I only pushed to staging and found it broken there

**Pull request merged**